packages/types/src/middleware.ts

```diff
@@ -360,10 +314,12 @@ export interface MiddlewareStack<Input extends object, Output extends object> {
   use(pluggable: Pluggable<Input, Output>): void;
 
   /**
-   * Create a shallow clone of this list. Step bindings and handler priorities
+   * Create a shallow clone of this stack. Step bindings and handler priorities
    * and tags are preserved in the copy.
+   * Users can optionally supply a base stack that will be merged into the clone
+   * output but its middleware will take preference.
    */
-  clone(): MiddlewareStack<Input, Output>;
+  clone(base?: MiddlewareStack<Input, Output>): MiddlewareStack<Input, Output>;
```

Comment:
Is there a requirement for merging stacks?

If yes, this should a separate operation, and not clone.

Reply:
Addressed in 1e15a4b